packages/headless/src/api/preprocess-request.ts

```diff
@@ -1,14 +1,21 @@
 import {AnalyticsClientOrigin} from 'coveo.analytics/dist/definitions/client/analyticsRequestClient';
+import {SearchApiClientMethod, SearchOrigin} from './search/search-metadata';
 
 export interface PlatformRequestOptions extends RequestInit {
   url: string;
 }
 
 export type PlatformClientOrigin = AnalyticsClientOrigin | 'searchApiFetch';
 
+export interface RequestMetadata {
+  clientMethod: SearchApiClientMethod;
+  origin?: SearchOrigin;
```

Comment:
A lot more can be added from different clients & origin.
The idea is that `clientMethod` is whatever method is called on the Api client, then `origin` is to differentiate the origin of this method when further differentiation is required